modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java

```diff
@@ -163,6 +166,9 @@ public List<AbstractCredentialInfo> getCredentials(
      */
     public void addCredential(CreateCredential createCredential) {
         checkInputCredentialIsValid(createCredential);
+        if (aesUtil.isEnabledAes()) {
```

Comment:
we can avoid the condition check here. 
instead just do this check within in the encode and decode method. 
We can make isEnabledAes() private